### PR TITLE
[M700]  Remove Case in summary sample event sql

### PR DESCRIPTION
**Pull request by gridcell**
Removing Case/When in the SummarySampleEventSQLModel meaning that all protocol summaries are being cached.

Add custom queryset so private data can be removed on the fly if the user doesn't have permission to view.

## Summary by CodeRabbit

- **New Features**
	- Enhanced data privacy with the introduction of a `privatize` method for more secure data handling.
	- Added a new `protocols` field to improve data serialization and representation.
	- Updated the summary sample event management system to support project-specific operations and foreground execution options.

- **Enhancements**
	- Improved the structure and logic of JSON field construction for better data management.

- **Tests**
	- Added comprehensive tests to validate user access levels to private data, ensuring robust data privacy measures.

- **Documentation**
	- Updated field names and query logic documentation for clarity and better understanding.

### CODE REVIEW

**Comment by gridcell** [comment]
[Remove Case in summary sample event sql](https://trello.com/c/6bq2pYNC/700-remove-case-in-summary-sample-event-sql)

**Comment by gridcell** [comment]
@coderabbitai review

File: src/tools/management/commands/update_summaries.py [hunk 1]
```diff
@@ -21,20 +21,44 @@ def add_arguments(self, parser):
...
+        for project in projects:
+            if in_foreground:
+                update_summary_cache(project_id=project.pk, skip_test_project=skip_test_project)
+            else:
+                submit_job(
+                    5,
+                    update_summary_cache,
+                    project_id=project.pk,
+                    skip_test_project=skip_test_project,
+                )
```
[the remark this answers is not in the document]
Comment (saanobhaai) [reply src/tools/management/commands/update_summaries.py:61]: @gridcell This doesn't seem like a bad idea...

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Wow, fancy... This looks good except for the minor coderabbit suggestion (which I would not have flagged).

**Pull request closed without merging**